core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -855,16 +858,17 @@ abstract class RDD[T: ClassTag](
    * a map on the other).
    */
   def zip[U: ClassTag](other: RDD[U]): RDD[(T, U)] = withScope {
-    zipPartitions(other, preservesPartitioning = false) { (thisIter, otherIter) =>
-      new Iterator[(T, U)] {
-        def hasNext: Boolean = (thisIter.hasNext, otherIter.hasNext) match {
-          case (true, true) => true
-          case (false, false) => false
-          case _ => throw new SparkException("Can only zip RDDs with " +
-            "same number of elements in each partition")
+    zipPartitionsInternal(other, preservesPartitioning = false, orderSensitiveFunc = true) {
+      (thisIter, otherIter) =>
+        new Iterator[(T, U)] {
+          def hasNext: Boolean = (thisIter.hasNext, otherIter.hasNext) match {
+            case (true, true) => true
+            case (false, false) => false
+            case _ => throw new SparkException("Can only zip RDDs with " +
+              "same number of elements in each partition")
+          }
+          def next(): (T, U) = (thisIter.next(), otherIter.next())
         }
-        def next(): (T, U) = (thisIter.next(), otherIter.next())
-      }
```

Comment:
Bulk of the change here is simply indentation right (except for zipPartitions -> zipPartitionsInternal and flag) ? I want to make sure I did not miss something here.

Reply:
yea, just indentation change.